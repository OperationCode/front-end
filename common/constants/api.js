import { get, post, patch } from 'common/utils/api-utils';

/* GET REQUESTS */
export const getCodeSchoolsPromise = () => get('code_schools');
export const getMentorPromise = id => get(`mentors/${id}`);
export const getMentorsPromise = () => get('mentors');
export const getRequestsPromise = () => get('requests');
export const getScholarshipPromise = id => get(`scholarships/${id}`);
export const getScholarshipsPromise = () => get('scholarships');
export const getServicesPromise = () => get('services');
<<<<<<< HEAD
export const getTeamMembersPromise = () => get('team_members');

/* POST REQUESTS */
export const postMentorRequestPromise = ({ language, additionalDetails, mentor, service, team }) =>
=======

/* POST REQUESTS */
export const postMentorRequestPromise = ({ language, additionalDetails, mentor, service }) =>
>>>>>>> 1e3818fe1ed61deeda96881e0e7a718d44c1dd4e
  post('requests', {
    request: {
      details: additionalDetails,
      language,
      requested_mentor_id: mentor,
      service_id: service,
<<<<<<< HEAD
      team:team

=======
>>>>>>> 1e3818fe1ed61deeda96881e0e7a718d44c1dd4e
    },
  });

/* PATCH REQUESTS */
export const patchUpdateMentorRequestPromise = ({ request, status, mentor }) =>
  patch(`requests/${request}`, {
    request: {
      status,
      mentor,
    },
  });
