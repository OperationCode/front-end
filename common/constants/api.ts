import { patch, post } from 'common/utils/api-utils';
import type { RegistrationFormValues } from 'components/Forms/RegistrationForm/RegistrationForm';
import type { MilitaryStatusFormShape } from 'components/Forms/UpdateProfileForm/steps/MilitaryStatus';
import type { ProfessionalDetailsFormShape } from 'components/Forms/UpdateProfileForm/steps/ProfessionalDetails';

export const registerUser = (values: RegistrationFormValues) =>
  post('/api/registration/new', values);

export const updateUser = (values: ProfessionalDetailsFormShape | MilitaryStatusFormShape) =>
  patch('/api/registration/update', values);
