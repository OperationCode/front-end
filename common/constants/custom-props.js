import PropTypes from 'prop-types';

/*
 * When declaring a new, commonly utilized prop type, please also export an object that can be used
 * by the Knobs Storybook addon easily. See Button's default story with analyticsObject for example.
 * NOTE: Export it as a function so that the object isn't mutable across multiple stories.
 */

/* eslint-disable import/prefer-default-export */
// See: https://github.com/react-ga/react-ga#reactgaeventargs for reference
export const googleAnalyticsEventPropType = PropTypes.shape({
  // A description of the behaviour. E.g. 'Clicked Delete', 'Added a component', 'Deleted account'
  action: PropTypes.string.isRequired,

  // A top level category for these events. E.g. 'User', 'Navigation', 'App Editing', etc.
  category: PropTypes.string.isRequired,

  // More precise labelling of the related action. E.g. alongside the 'Added a component' action,
  // we could add the name of a component as the label. E.g. 'Survey', 'Heading', 'Button', etc.
  label: PropTypes.string,

  // A means of recording a numerical value against an event. E.g. a rating, a score, etc.
  value: PropTypes.number,

  // If an event is not triggered by a user interaction, but instead by our code (e.g. on page
  // load, it should be flagged as a nonInteraction event to avoid skewing bounce rate data.
  nonInteraction: PropTypes.boolean,

  // This specifies the transport mechanism with which hits will be sent. Valid values include
  // 'beacon', 'xhr', or 'image'.
  transport: PropTypes.oneOf(['beacon', 'xhr', 'image']),
});

export const googleAnalyticsEventStoryObjectFactory = () => ({
  action: '',
  category: '',
  label: '',
  value: undefined,
  nonInteraction: false,
  transport: undefined,
});
