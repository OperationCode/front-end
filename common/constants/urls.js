/* eslint-disable max-len */
export const s3 = 'https://s3.amazonaws.com/operationcode-assets/';
export const donateLink = 'https://secure.lglforms.com/form_engine/s/BRtP7QUKyHOyEYsZROsRew';
export const leadershipCircleLink =
  'https://secure.lglforms.com/form_engine/s/L428AQ2rrsFJQyy5Fbglvg';
