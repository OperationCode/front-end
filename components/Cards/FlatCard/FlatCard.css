.FlatCard {
  box-sizing: border-box;
  margin: 1.5rem 1rem;
  padding-top: 60px;
  position: relative;
  max-width: 400px;
}

.image {
  order: 1;
  height: 200px;
  width: 200px;
  position: absolute;
  top: 0;
}

.rowCenter {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.header {
  order: 2;
  text-align: center;
}

.content {
  order: 3;
}

.borderContainer {
  padding: 150px 2rem 2rem;
  border: 3px solid var(--primary);
}

.divider {
  width: 85%;
  border: 1px solid var(--primary);
}
