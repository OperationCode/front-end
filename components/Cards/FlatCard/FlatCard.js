import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './FlatCard.css';

FlatCard.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  header: PropTypes.node,
  imageAlt: PropTypes.string.isRequired,
  imageSource: PropTypes.string.isRequired,
};

FlatCard.defaultProps = {
  className: undefined,
  header: undefined,
};

function FlatCard({ children, className, header, imageSource, imageAlt }) {
  return (
    <article className={classNames(styles.FlatCard, className)}>
      <div className={styles.borderContainer}>
        {header && <div className={styles.header}>{header}</div>}
        <div className={styles.rowCenter}>
          <img className={styles.image} src={imageSource} alt={imageAlt} />
        </div>
        {header && <hr className={styles.divider} />}
        <div className={styles.children}>{children}</div>
      </div>
    </article>
  );
}

export default FlatCard;
