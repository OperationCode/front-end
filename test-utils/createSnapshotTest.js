/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
