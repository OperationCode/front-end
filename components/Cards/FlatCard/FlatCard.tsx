import classNames from 'classnames';
import { twMerge } from 'tailwind-merge';
import Image from 'next/legacy/image';
import { FLAT_CARD_IMAGE } from '@/common/constants/testIDs';
import { getPlaceholder } from '@/common/utils/next-utils';
import { ReactNode } from 'react';

interface FlatCardProps {
  button?: ReactNode;
  children: ReactNode;
  className?: string;
  header?: ReactNode;
  image?: {
    source: string;
    alt: string;
  };
}

export function FlatCard({ button: Button, children, className, header, image }: FlatCardProps) {
  const hasImage = image && image.source;

  return (
    <article
      className={classNames('box-border my-6 mx-4 relative max-w-[400px]', className, {
        'pt-20 sm:pt-14': hasImage,
      })}
    >
      <div
        className={twMerge(
          'p-8 flex flex-col gap-5 border-4 border-solid border-themePrimary',
          hasImage && `pt-32`,
        )}
      >
        {header && <div className="text-center">{header}</div>}

        {hasImage && (
          <div className="flex justify-center -my-5">
            <div
              data-testid={FLAT_CARD_IMAGE}
              className="absolute -top-4 border-solid border-themePrimary border-[3px] w-[194px] h-[194px]"
            >
              <Image
                src={image.source}
                alt={image.alt ?? ''}
                layout="fill"
                objectFit="cover"
                placeholder="blur"
                blurDataURL={getPlaceholder(194, 194)}
              />
            </div>
          </div>
        )}

        {header && <hr className="w-full border-2 border-solid border-themePrimary" />}

        <div>{children}</div>

        {Button && (
          <div className="text-center absolute left-0 right-0 -bottom-7 [&>button]:hover:bg-white [&>button]:hover:text-themeSecondary [&>button]:focus-visible:text-themePrimary [&>button]:focus-visible:bg-white [&>a]:hover:bg-white [&>a]:hover:text-themeSecondary [&>a]:focus-visible:bg-white [&>a]:focus-visible:text-themeSecondary ">
            {Button}
          </div>
        )}
      </div>
    </article>
  );
}
