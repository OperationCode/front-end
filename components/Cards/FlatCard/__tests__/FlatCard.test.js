/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import { s3 } from 'common/constants/urls';
import FlatCard from '../FlatCard';

describe('FlatCard', () => {
  const requiredProps = {
    imageSource: `${s3}headshots/david_molina.jpg`,
    imageAlt: "David Molina's face",
    children: (
      <>
        <p>Example content goes here</p>
        <p>More content goes here</p>
      </>
    ),
  };

  it('should render with required props', () => {
    createSnapshotTest(<FlatCard {...requiredProps} />);
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(
      <FlatCard
        {...requiredProps}
        className="test-class"
        header={
          <>
            <h1>Main heading</h1>
            <h6>Sub heading</h6>
          </>
        }
      />,
    );
  });

  it('should not render a horizontal ruler when header is undefined', () => {
    const wrapper = shallow(<FlatCard {...requiredProps} header={undefined} />);

    expect(wrapper.find('hr').exists()).toStrictEqual(false);
  });

  it('should render a horizontal ruler when header is passed', () => {
    const wrapper = shallow(<FlatCard {...requiredProps} header={<h1>Howdy!</h1>} />);

    expect(wrapper.find('hr').exists()).toStrictEqual(true);
  });
});
