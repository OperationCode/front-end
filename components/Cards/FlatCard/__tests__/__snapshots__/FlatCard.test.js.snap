// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatCard should render with many props assigned 1`] = `
<article
  className="FlatCard test-class"
>
  <div
    className="borderContainer"
  >
    <div
      className="header"
    >
      <h1>
        Main heading
      </h1>
      <h6>
        Sub heading
      </h6>
    </div>
    <div
      className="rowCenter"
    >
      <img
        alt="David Molina's face"
        className="image"
        src="https://s3.amazonaws.com/operationcode-assets/headshots/david_molina.jpg"
      />
    </div>
    <hr
      className="divider"
    />
    <div
      className="children"
    >
      <p>
        Example content goes here
      </p>
      <p>
        More content goes here
      </p>
    </div>
  </div>
</article>
`;

exports[`FlatCard should render with required props 1`] = `
<article
  className="FlatCard"
>
  <div
    className="borderContainer"
  >
    <div
      className="rowCenter"
    >
      <img
        alt="David Molina's face"
        className="image"
        src="https://s3.amazonaws.com/operationcode-assets/headshots/david_molina.jpg"
      />
    </div>
    <div
      className="children"
    >
      <p>
        Example content goes here
      </p>
      <p>
        More content goes here
      </p>
    </div>
  </div>
</article>
`;
